### ISSUE TITLE
[Bug]: attributes updates are not applied to ReactNodeViewRenderer element

### ISSUE THREAD

**Issue by programateus** [ueberdosis/tiptap#4239]
### Which packages did you experience the bug in?

react

### What Tiptap version are you using?

2.0.4

### What’s the bug you are facing?

I'm trying to extend the table-cell extension so I can show a button with a menu tooltip on the bottom left of the cell. I was able to implement the said feature, but when I use the merge cell command, the colspan of the element is not updated, I have tried passing the attributes manually to the ReactNodeViewRenderer options but it did not worked:

```jaavascript
addNodeView() {
  return (props) =>
    ReactNodeViewRenderer(TableCellView, {
      as: "td",
      attrs: props.node.attrs
    })(props);
}
```

I also tried updating attributes inside the custom component. but it also not worked:

```javascript
useEffect(() => {
  updateAttributes({
    colspan: node.attrs.colspan,
    rowspan: node.attrs.rowspan,
    colwidth: node.attrs.colwidth
  });
}, [
  updateAttributes,
  node.attrs.colspan,
  node.attrs.rowspan,
  node.attrs.colwidth
]);
```

To reproduce, just click on `Insert Table`, select two cells and click on `Merge Cells` on the provided reproduction example.

### What browser are you using?

Firefox

### Code example

https://codesandbox.io/s/tiptap-table-issue-tlwcnt

### What did you expect to happen?

The table cell should update its attributes and show the correct width of the cell on the table.

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by programateus** [comment]
Just found a workaround and possibly a fix after copying the whole @tiptap/react package to my project and adding the following code to ReactNodeView@update:

```javascript
update(node: ProseMirrorNode, decorations: DecorationWithType[]) {
  const updateProps = (props?: Record<string, any>) => {
    this.renderer.updateProps(props);
  };

  if (node.type !== this.node.type) {
    return false;
  }

  if (typeof this.options.update === "function") {
    const oldNode = this.node;
    const oldDecorations = this.decorations;

    this.node = node;
    this.decorations = decorations;

    return this.options.update({
      oldNode,
      oldDecorations,
      newNode: node,
      newDecorations: decorations,
      updateProps: () => updateProps({ node, decorations }),
    });
  }

  if (node === this.node && this.decorations === decorations) {
    return true;
  }

  // ------------------ Added this line ---------------------
  Object.keys(node.attrs).forEach((key) => {
    this.renderer.element.setAttribute(key, node.attrs[key]);
  });
  // --------------------------------------------------------
  this.node = node;
  this.decorations = decorations;

  updateProps({ node, decorations });

  return true;
}
```

If this could be a fix I would be happy to make a pull request.

**Comment by arpit016** [comment]
@programateus I am running into the same issue. Would like to have a proper solution for this one.

**Comment by programateus** [comment]
@arpit016 you have to use everything from the local copy of tiptap/react. Make sure you are not using the @tiptap/react package from the node_modules folder.

**Comment by sarasilvaworkera** [comment]
I'm also running into this issue :(

**Comment by bdbch** [comment]
Could you open a PR with your changes @programateus since you already started working on it?

**Comment by sarasilvaworkera** [comment]
Not sure if this could be fixed in this scope or if it's too unrelated, but is it possible to add some kind of attribute regarding the cell position in the table (that gets updated every time there are row/column changes)? I need this kind of information to add some custom behavior to the cell...

**Comment by tarun-29** [comment]
hi @programateus i did what you exactly said and it's working fine when i implemented in codesandbox but when i tried to do the same in my localhost it's not working at all basically inside the ReactNodeViewRenderer `get dom()` function i am getting `Uncaught runtime errors:` `TypeError: Cannot read properties of undefined (reading 'element')` and also one more interesting thing i found out i had cloned you repo which is `marknotes-main` and i had used exactly same code which i am using in my simple react app and the code is working on your app but it's not working on my react app and giving same error `Uncaught runtime errors:` `TypeError: Cannot read properties of undefined (reading 'element')` Any guidance will be helpful

**Comment by tarun-29** [comment]
hi @programateus  Here is screenshot for the above error, i am trying this in fresh `create-react-app` project and not next.js
<img width="1055" alt="Screenshot 2023-08-23 at 1 39 29 PM" src="https://github.com/ueberdosis/tiptap/assets/48859385/ab4b19f7-a274-4b98-bb26-9a447027923a">

**Comment by tarun-29** [comment]
hi @programateus thanks for the solution it works like magic, although i have one question i earlier used patch-package but i was doing it in single file which is ReactNodeViewRenderer how did you figure out that we have to do the same in `react/dist` directory, will be helpful if you can give me some reason to understand the necessity of this folder

**Comment by programateus** [comment]
I would say thats beause it uses the compiled code instead of the source code, I edited everything just to make sure it would work. Sometimes you have to remove `node_modules` and install everything again to work too.

**Comment by duxphp** [comment]
I'm having the same problem, but the bug hasn't been updated yet

**Comment by nperez0111** [comment]
This is released with v2.7.0 https://github.com/ueberdosis/tiptap/releases/tag/v2.7.0
